File: src/main/java/com/mattchowning/file_read_write/SharedConstants.java
package com.mattchowning.file_read_write;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.nio.charset.Charset;

import io.netty.util.CharsetUtil;

public class SharedConstants {

    public static final String OAUTH_PATH = "/oauth";
    public static final String GRANT_TYPE_KEY = "grant_type";
    public static final String GRANT_TYPE_PASSWORD = "password";
    public static final String GRANT_TYPE_REFRESH_TOKEN = "refresh_token";
    public static final String PASSWORD_KEY = "password";
    public static final String USERNAME_KEY = "username";
    public static final String REFRESH_TOKEN_KEY = "refresh_token";
    public static final Charset RESPONSE_CHARSET = CharsetUtil.UTF_8;
    public static final String FILE_HOST = "localhost";
    public static final int FILE_PORT = 8081;
    public static final String OAUTH_HOST = FILE_HOST;
    public static final int OAUTH_PORT = 8080;

    public static final Gson GSON = new GsonBuilder()
            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
            .create();
}

File: src/main/scala/com/mattchowning_scala/file_read_write/SharedConstants.scala
package com.mattchowning_scala.file_read_write

import com.google.gson._
import java.nio.charset.Charset
import io.netty.util.CharsetUtil

// TODO turn into trait?
object SharedConstants {
  val OAUTH_PATH: String = "/oauth"
  val GRANT_TYPE_KEY: String = "grant_type"
  val GRANT_TYPE_PASSWORD: String = "password"
  val GRANT_TYPE_REFRESH_TOKEN: String = "refresh_token"
  val PASSWORD_KEY: String = "password"
  val USERNAME_KEY: String = "username"
  val REFRESH_TOKEN_KEY: String = "refresh_token"
  val RESPONSE_CHARSET: Charset = CharsetUtil.UTF_8
  val FILE_HOST: String = "localhost"
  val FILE_PORT: Int = 8081
  val OAUTH_HOST: String = FILE_HOST
  val OAUTH_PORT: Int = 8080
  val GSON: Gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create
}